src/librustdoc/clean/utils.rs

```diff
@@ -331,11 +334,27 @@ crate fn print_evaluated_const(tcx: TyCtxt<'_>, def_id: DefId) -> Option<String>
 
 fn format_integer_with_underscore_sep(num: &str) -> String {
     let num_chars: Vec<_> = num.chars().collect();
```

Comment:
A good follow-up would be to remove `num_chars` altogether - `num` is always ascii so you should be able to just use bytes directly. Doesn't need to be in this PR though :)